src/main/java/spoon/support/reflect/declaration/CtClassImpl.java

```diff
@@ -240,4 +254,35 @@ public void replace(CtStatement element) {
 	public CtClass<T> clone() {
 		return (CtClass<T>) super.clone();
 	}
+
+	@Override
+	public Object newInstance() {
+		try {
+			JDTBatchCompiler compiler = new JDTBatchCompiler(null, false) {
+				@Override
+				public CompilationUnit[] getCompilationUnits() {
+					CompilationUnit[] units = new CompilationUnit[1];
+					units[0] = new CompilationUnitWrapper(CtClassImpl.this);
+					return units;
+				}
+			};
+			String[] args = new String[] {};
+			args = new JDTBuilderImpl() //
+					.classpathOptions(new ClasspathOptions().classpath().binaries(getFactory().getEnvironment().getBinaryOutputDirectory())) //
+					.complianceOptions(new ComplianceOptions().compliance(getFactory().getEnvironment().getComplianceLevel())) //
+					.advancedOptions(new AdvancedOptions().preserveUnusedVars().continueExecution()) //
+					.annotationProcessingOptions(new AnnotationProcessingOptions().compileProcessors()) //
+					.sources(new SourceOptions().sources()) //
+					.build();
+			compiler.compile(args);
+			Class<?> klass = new URLClassLoader(new URL[] { new File("./").toURL() }).loadClass(getQualifiedName());
+
+			// cleaning afterwards
+			new File(getFactory().getEnvironment().getBinaryOutputDirectory() + "/" + getQualifiedName().replace('.', '/') + ".class").delete();
+
+			return klass.newInstance();
```

Comment:
I think it is a good idea to force the compilation the complete project because if the user did not set shouldCompile to true, all reference to other classes of the project will produce compilation error.
Same error if the klass contains a reference to a new class created with spoon.